internal/telemetry/service_test.go

```diff
@@ -2,14 +2,49 @@ package telemetry
 
 import (
 	"errors"
+	"io/ioutil"
+	"log"
+	"os"
 	"testing"
 
 	"github.com/10gen/realm-cli/internal/utils/test/assert"
 )
 
 func TestNewService(t *testing.T) {
 	t.Run("Should create the expected Service", func(t *testing.T) {
-		service := NewService(ModeOn, "userID", "command")
+		service := NewService(ModeStdout, "userID", nil, "command")
+
+		assert.Equal(t, "command", service.command)
+		assert.True(t, service.executionID != "", "service execution id must not be blank")
+		assert.Equal(t, "userID", service.userID)
+		assert.NotNil(t, service.tracker)
+	})
+
+	createSegmentFn := func() *Service {
```

Comment:
to tack on with the `testServiceOutput` refactor I suggested below, I'll channel my inner-Tim here and resist the one-liner helper func.  instead, we can have:

```go
const (
  testUser = "userID"
  testCommand = "command"
)

var (
  testLogger = log.New(os.Stdout, "LogPrefix ", log.Lmsgprefix)
)

// and instead of passing `createSegmentFn` around here, just instantiate a `*Service` in the tests directly and use those

// and if you *really* want a helper for this still, then go with:
func newService(mode Mode) *Service {
  return NewService(mode, testUser, testLogger, testCommand) // then make sure to be using this for all `NewService` calls
}
```